### ISSUE TITLE
allow rich description rendering in plugin admin page

### ISSUE THREAD

**Issue by McFoggy** [gitbucket/gitbucket#1287]
**Problem description**:
Promotion of plugins is not very easy and allowing to render rich text in the plugin page would help:

- users to understand the plugin feature
- maintainers to promote their plugin

Instead of simple text rendering, we could allow other formats (markdown, ...)

![image](https://cloud.githubusercontent.com/assets/1119660/18229864/dfd312de-7288-11e6-98dc-655565fc8f71.png)

**Comment by eprozium** [comment]
@McFoggy a little long description and richtext would definitely help.

Some other fields that could help:

 - url of the plug-in: users could read there further details and also see screenshots.
 - GitBucket version the plug-in is compatible with (to know if it needs upgrade or not).
 - Author : to promote plug creation.

**Comment by takezoe** [comment]
Url and author should be included in plug-in information. However version compatibility should be known before install plug-in.

My opinion is plug-in developers should write README in their GitHub repository. Then GitBucket (and gitbucket community plugins) can retrieve plug-in description from there.

**Comment by eprozium** [comment]
>  However version compatibility should be known before install plug-in.

Yes, you are right. It needs to be known, but it can also be displayed. Together with the information about the updates (about GitBucket and plug-ins), the user can decide to do an update or wait since an update would break things, or not all plug-ins work yet.

> My opinion is plug-in developers should write README in their GitHub repository. Then GitBucket (and gitbucket community plugins) can retrieve plug-in description from there.

The Readme should be of course mandatory, but that should be for human "parsing" only. For the software a more proper solution would be a mandatory descriptor file like IntelliJ has for plug-ins (but not XML, but maybe JSON, or even better EDN http://mandubian.com/2014/12/21/scaledn/ )

This way, more things could be automated:

 - listing (and searching) of plug-ins on the plugin community site.
 - listing (and searching) of plug-ins from GitBucket itself (if the installation has internet access - not all have)
 - automatic installation (using tag release)
 - automatic plug-in check and update (based on the compatibility version)
